exports/taskfiles/utils/boost.yaml

```
@@ -1,128 +1,127 @@
 version: "3"
 
 includes:
+  checksum: "checksum.yaml"
   remote: "remote.yaml"
 
 set: ["u", "pipefail"]
 shopt: ["globstar"]
 
 tasks:
-  # Runs the bootstrap.sh generate step in the given source directory. Boost only supports
-  # in-source generation and building.
+  # Copy the source directory into the generate directory. Runs the `bootstrap.sh` script to
+  # generate `b2` and `project-config.jam` files.
   #
   # @param {string} SOURCE_DIR Project source directory.
+  # @param {string} GENERATE_DIR Directory in which to generate the build files.
+  # @param {string} GENERATE_CHECKSUM Checksum file of the `GENERATE_DIR` directory.
   # @param {string} INSTALL_PREFIX Path prefix of where the project should be installed.
   # @param {string[]} TARGETS Target libraries to build.
   # @param {string[]} [EXTRA_ARGS] Any additional arguments to pass to the generate command.
   generate:
     internal: true
-    dir: "{{.SOURCE_DIR}}"
     vars:
       EXTRA_ARGS:
         ref: "default (list) .EXTRA_ARGS"
     requires:
-      vars: ["SOURCE_DIR", "INSTALL_PREFIX", "TARGETS"]
+      vars: ["SOURCE_DIR", "GENERATE_DIR", "GENERATE_CHECKSUM", "INSTALL_PREFIX", "TARGETS"]
+    sources:
+      - "{{.SOURCE_DIR}}/**/*"
+    generates:
+      - "{{.GENERATE_CHECKSUM}}"
+    deps:
+      - task: "checksum:validate"
+        vars:
+          CHECKSUM_FILE: "{{.GENERATE_CHECKSUM}}"
+          INCLUDE_PATTERNS:
+            - "{{.GENERATE_DIR}}"
     cmds:
       - >-
+        rm -rf "{{.GENERATE_DIR}}"
+      - >-
+        mkdir -p "{{.GENERATE_DIR}}"
+      - >-
+        cp -R "{{.SOURCE_DIR}}"/* "{{.GENERATE_DIR}}/"
```

I think this is equivalent, right?

```suggestion
        cp -R "{{.SOURCE_DIR}}" "{{.GENERATE_DIR}}"
```

---

No. If the target directory exists, this will copy the source inside target, so it will become `target/source`.